drivers/input/input_stmpe811.c

```diff
@@ -0,0 +1,469 @@
+
+	/* Start polling to detect release */
+	data->poll_release_counter = 0;
+	k_work_schedule(&data->polling_work, K_USEC(data->poll_period_us));
+
+	return 0;
+}
+
+static void stmpe811_work_handler(struct k_work *work)
+{
+	struct stmpe811_data *data = CONTAINER_OF(work, struct stmpe811_data, work);
+
+	stmpe811_process(data->dev);
+}
+
+static void stmpe811_poll_release_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *polling_work = CONTAINER_OF(work, struct k_work_delayable, work);
+	struct stmpe811_data *data = CONTAINER_OF(polling_work, struct stmpe811_data, polling_work);
+
+	const struct device *dev = data->dev;
+	const struct stmpe811_config *config = data->dev->config;
+
+	bool poll_again = true;
+	int err;
+	uint32_t current_cycles;
+	uint32_t cycles_diff;
+	uint32_t wait_period;
+
+	k_timepoint_t poll_time_end = sys_timepoint_calc(K_USEC(data->poll_timeout_us));
+	uint32_t start_period_cycles = k_cycle_get_32();
+	bool touch = stmpe811_ts_detect_touch(dev);
+
+	/* Clear FIFO - data->poll_period_us has to be high enough
+	 * to fill it up before next
+	 */
+
+	err = i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG, FIFO_STA_CLEAR);
+	err += i2c_reg_write_byte_dt(&config->bus, STMPE811_FIFO_STA_REG,
+						FIFO_STA_OPERATIONAL);
+
+	if (err < 0) {
+		LOG_ERR("Could not clear FIFO after polling");
+	} else {
+		if (!touch) {
+			data->poll_release_counter++;
+		}
+
+		if (data->poll_release_counter >= data->poll_release_threshold ||
+			sys_timepoint_expired(poll_time_end)) {
+			input_report_key(data->dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+			stmpe811_clear_global_it(dev, STMPE811_TS_IT);
+			poll_again = false;
+		}
+
+		/* Subtract the time invested from the sleep period */
+		current_cycles = k_cycle_get_32();
+		cycles_diff = current_cycles - start_period_cycles;
+		wait_period = data->poll_period_us - k_cyc_to_us_floor32(cycles_diff);
+
+		/* Override wait_period in case it is less than 1 ms */
+		if (wait_period < USEC_PER_MSEC) {
+			wait_period = USEC_PER_MSEC;
+		}
+
+		if (wait_period > data->poll_period_us) {
+			wait_period = data->poll_period_us;
+		}
+
+		if (poll_again) {
+			k_work_schedule(&data->polling_work, K_USEC(wait_period));
+		}
+	}
+}
+
+static void interrupt_handler(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	LOG_DBG("stmpe811 generated an interrupt");
+	struct stmpe811_data *data = CONTAINER_OF(cb, struct stmpe811_data, int_gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static int stmpe811_verify_chip_id(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	uint8_t buf[2];
+
+	i2c_burst_read_dt(&config->bus, STMPE811_CHP_ID_LSB_REG, buf, 2);
+
+	if (((buf[0] << 8) | buf[1]) != CHIP_ID) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stmpe811_init(const struct device *dev)
+{
+	const struct stmpe811_config *config = dev->config;
+	struct stmpe811_data *data = dev->data;
+
+	if (!device_is_ready(config->bus.bus)) {
```

Comment:
You can use `i2c_is_ready_dt(&config->bus)` here

Reply:
Done